File: factory/go/hero.go
package main

type hero struct {
	name         string
	agility      int
	strength     int
	intelligence int
}

func (h *hero) setName(name string) {
	h.name = name
}

func (h *hero) getName() string {
	return h.name
}

func (h *hero) setAgility(agility int) {
	h.agility = agility
}

func (h *hero) getAgility() int {
	return h.agility
}

func (h *hero) setStrength(strength int) {
	h.strength = strength
}

func (h *hero) getStrength() int {
	return h.strength
}

func (h *hero) setIntelligence(intelligence int) {
	h.intelligence = intelligence
}

func (h *hero) getIntelligence() int {
	return h.intelligence
}

File: java/src/factory/Hero.java
package factory;

public class Hero {
  private String name;
  private int agi;
  private int str;

  public Hero(String name, int agi, int str){
    this.name = name;
    this.agi = agi;
    this.str = str;
  }

  public String getName() {
    return name;
  }

  public int getAgi() {
    return agi;
  }

  public int getStr() {
    return str;
  }
}
